src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -805,7 +805,7 @@ void maybeDeleteRecordedAutoDeleteExchange(String exchange) {
                     // last binding where this exchange is the source is gone, remove recorded exchange
                     // if it is auto-deleted. See bug 26364.
                     if((x != null) && x.isAutoDelete()) {
-                        this.recordedExchanges.remove(exchange);
+                        deleteRecordedExchange(exchange);
```

Comment:
It was possible to have abandoned e2e bindings in the recordedBindings.

Workflow to reproduce:
1) create durable topic exchange -> auto-delete headers exchange -> auto-delete queue -> consumer
2) cancel consumer
3) calls maybeDeleteRecordedAutoDeleteQueue which ends up deleting the recorded queue
4) calls maybeDeleteRecordedAutoDeleteExchange and removes the recorded AD headers exchange

In that case the e2e binding from the durable topic exchange to the now deleted headers exchange still existed.